GVFS/GVFS.Common/Cleanup/GitCleanupQueue.cs

```diff
@@ -0,0 +1,103 @@
+﻿using GVFS.Common.Tracing;
+using System;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Threading;
+
+namespace GVFS.Common.Cleanup
+{
+    public class GitCleanupQueue
+    {
+        private GVFSContext context;
+        private BlockingCollection<GitCleanupStep> queue = new BlockingCollection<GitCleanupStep>();
+        private CancellationTokenSource cancellationToken = new CancellationTokenSource();
+        private GitCleanupStep currentStep;
+
+        public GitCleanupQueue(GVFSContext context)
+        {
+            this.context = context;
+            Thread cleanupWorker = new Thread(() => this.RunQueue());
+            cleanupWorker.Name = "CleanupWorker";
+            cleanupWorker.Start();
+        }
+
+        public void Enqueue(GitCleanupStep step)
+        {
+            this.queue.Add(step);
+        }
+
+        public void Stop()
+        {
+            this.cancellationToken.Cancel();
+            this.currentStep?.Stop();
+        }
+
+        /// <summary>
+        /// This method is public for test purposes only.
+        /// </summary>
+        public bool EnlistmentRootReady()

```

Comment:
Can this be added to the `MockFileSystem` instead of the production code?

Reply:
We just need to update the comment.  It's needed for production code to verify the device is ready

Reply:
This is necessary logic to check if the drive is locked due to BitLocker or an unplugged USB drive. We need it in production and only make it public so I could write unit tests.